camus-etl-kafka/src/test/java/com/linkedin/camus/etl/kafka/CamusJobTestWithMock.java

```diff
@@ -193,25 +208,117 @@ private void setupJob1() {
     ByteBufferMessageSet messageSet = new ByteBufferMessageSet(messages);
     EasyMock.expect(fetchResponse.messageSet(EasyMock.anyString(), EasyMock.anyInt())).andReturn(messageSet).times(1);
     _mocks.add(fetchResponse);
-    
-    // For SimpleConsumer
-    _simpleConsumer = EasyMock.createMock(SimpleConsumer.class);
+    return fetchResponse;
+  }
+
+  private void mockSimpleConsumer(TopicMetadataResponse metadataResponse, OffsetResponse offsetResponse,
+      FetchResponse fetchResponse) {
+    _simpleConsumer = EasyMock.createNiceMock(SimpleConsumer.class);
     _mocks.add(_simpleConsumer);
     EasyMock.expect(_simpleConsumer.send((TopicMetadataRequest)EasyMock.anyObject())).andReturn(metadataResponse).times(1);
-    EasyMock.expect(_simpleConsumer.getOffsetsBefore((OffsetRequest)EasyMock.anyObject())).andReturn(offsetResponse).times(2);
+    EasyMock.expect(_simpleConsumer.getOffsetsBefore((OffsetRequest)EasyMock.anyObject())).andReturn(offsetResponse).anyTimes();
```

Comment:
For the same reason, I think it shouldn't be an argument. We can have different methods, e.g., mockSimpleConsumer(), mockSimpleConsumerThrowsExceptionForOffsetRangeCall(), etc.